src/kernels/composable_kernel/include/kernel_algorithm/gridwise_convolution_backward_weights_implicit_gemm_v4r4_ncdhw_kczyx_nkdhw.hpp

```diff
@@ -0,0 +1,180 @@
+          index_t GemmMPerThread,
+          index_t GemmNPerThread,
+          index_t GemmKPerThread,
+          index_t GemmMLevel0Cluster,
+          index_t GemmNLevel0Cluster,
+          index_t GemmMLevel1Cluster,
+          index_t GemmNLevel1Cluster,
+          index_t ThreadGemmDataPerRead_GemmM,
+          index_t ThreadGemmDataPerRead_GemmN,
+          typename GemmABlockCopyThreadSliceLengths_GemmK_GemmM,
+          typename GemmABlockCopyThreadClusterLengths_GemmK_GemmM,
+          index_t GemmABlockCopySrcDataPerRead_GemmK,
+          index_t GemmABlockCopyDstDataPerWrite_GemmM,
+          typename GemmBBlockCopyThreadSliceLengths_GemmK_GemmN,
+          typename GemmBBlockCopyThreadClusterLengths_GemmK_GemmN,
+          index_t GemmBBlockCopySrcDataPerRead_GemmN,
+          index_t GemmBBlockCopyDstDataPerWrite_GemmN,
+          index_t GemmCThreadCopyDstDataPerWrite_GemmN1>
+struct GridwiseConvolutionBackwardWeightsImplicitGemm_v4r4_ncdhw_kczyx_nkdhw
+{
+    __device__ void Run(const Float* const __restrict__ p_in_global,
+                        const Float* const __restrict__ p_out_global,
+                        Float* const __restrict__ p_wei_global) const
+    {
+        constexpr auto I1 = Number<1>{};
+        constexpr auto I2 = Number<2>{};
+        constexpr auto I4 = Number<4>{};
+
+        constexpr auto in_n_c_di_hi_wi_global_desc  = InGlobalDesc{};
+        constexpr auto wei_k_c_z_y_x_global_desc    = WeiGlobalDesc{};
+        constexpr auto out_n_k_do_ho_wo_global_desc = OutGlobalDesc{};
+
+        constexpr index_t N  = in_n_c_di_hi_wi_global_desc.GetLengths()[0];
+        constexpr index_t C  = in_n_c_di_hi_wi_global_desc.GetLengths()[1];
+        constexpr index_t Di = in_n_c_di_hi_wi_global_desc.GetLengths()[2];
+        constexpr index_t Hi = in_n_c_di_hi_wi_global_desc.GetLengths()[3];
+        constexpr index_t Wi = in_n_c_di_hi_wi_global_desc.GetLengths()[4];
+
+        constexpr index_t K  = out_n_k_do_ho_wo_global_desc.GetLengths()[1];
+        constexpr index_t Do = out_n_k_do_ho_wo_global_desc.GetLengths()[2];
+        constexpr index_t Ho = out_n_k_do_ho_wo_global_desc.GetLengths()[3];
+        constexpr index_t Wo = out_n_k_do_ho_wo_global_desc.GetLengths()[4];
+
+        constexpr index_t Z = wei_k_c_z_y_x_global_desc.GetLengths()[2];
+        constexpr index_t Y = wei_k_c_z_y_x_global_desc.GetLengths()[3];
+        constexpr index_t X = wei_k_c_z_y_x_global_desc.GetLengths()[4];
+
+        constexpr index_t ConvStrideD = ConvStrides{}[0];
+        constexpr index_t ConvStrideH = ConvStrides{}[1];
+        constexpr index_t ConvStrideW = ConvStrides{}[2];
+
+        constexpr index_t ConvDilationD = ConvDilations{}[0];
+        constexpr index_t ConvDilationH = ConvDilations{}[1];
+        constexpr index_t ConvDilationW = ConvDilations{}[2];
+
+        // sanity-check for vectorized memory load
+        static_assert((Wo == 1 || (ConvStrideW == 1 || GemmBBlockCopySrcDataPerRead_GemmN == 1)) &&
```

Comment:
this check is for fwd, but it's not correct for bwd-wrw.

should be GemmBBlockCopySrcDataPerRead_GemmK everywhere. Please correct it or remove it. If removing, add ```todo``` here

Reply:
please wait 1-2 days, I already change read by GemmK,
I think we can merge at this PR and add this constrain